### PR TITLE
Generate documentation with jsdoc3, first draft.

### PR DESCRIPTION
**Pull request by nilguncelik**
I am creating the pull request now, but we talked that it will be merged once travis and gulp scripts are ready.

### CODE REVIEW

**Comment by nilguncelik** [comment]
I have added .travis.yml, gulpfile.js and package.json files.
I also rebased the "fix-documentation-formatting" branch on this one, to unify all commits on jsdoc generation.

We still need to make changes to travis.yml :
1. Set GH_TOKEN variable
   https://help.github.com/articles/creating-an-access-token-for-command-line-use/
2. Set secure key
  "gem install travis" and GH_TOKEN=<secret token here>
3. Globally install jsdoc ? Because it is not included in package.json. But didn't see it elsewhere in other example files.
4. Init submodule for jsdoc template jaguarjs-jsdoc?

File: package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "TartJS",
```
Comment (dashersw) [inline package.json]: We use lowercase for the first t, it's tartJS.

**Comment by dashersw** [comment]
@nilguncelik I believe we need to set `#{process.env.GH_TOKEN}` for the travis.yml, and set GH_TOKEN in travis configuration on their site. Exposing the key in our source code is not meaningful.

File: .gitmodules [hunk 2]
```diff
@@ -7,3 +7,6 @@
 [submodule "third_party/goog"]
 	path = third_party/goog
 	url = https://github.com/tart/closure-library.git
+[submodule "third_party/jaguarjs-jsdoc"]
```
Comment (dashersw) [inline .gitmodules]: We should find another way of including this in our projects. if travis allows bash scripts, we could just clone it. Or we could publish it to npm, and then just do an `npm install`; declaring `jaguarjs-jsdoc` as one of our `devDependencies`

**Comment by fatihacet** [comment]
@nilguncelik for @dashersw's [this comment](https://github.com/tart/tartJS/pull/206#issuecomment-125510807) see [this](http://docs.travis-ci.com/user/environment-variables/#Encrypted-Variables) link.

**Comment by nilguncelik** [comment]
@fatihacet thanks for the link :)

As far as I understand there are two ways to configure secret environment variables with travis.

One is to install travis gem and run the following command:
```bash
gem install travis
travis encrypt GH_TOKEN=super_secret_gh_token_here  --add env.global
```
This will generate the lines: 
```
env:
   global:
      - secure: <encrypted_key_here>
```
Then Travis will internally decrypt the secure variable and will know the value of `GH_TOKEN`. It will replace `${GH_TOKEN}` with that value whenever it encounters it.


The second way is to use `${process.env.GH_TOKEN}` in travis.yml, and set `GH_TOKEN` in travis configuration on their site as a secret environment variable as @dashersw suggests. 

@dashersw How do you think we should proceed? Also if we go with the first option, can you generate a secret key and send to me. I think the token I generate wont work as I dont have write access to the repo.

File: .gitmodules (hunk 2 above)
Comment (nilguncelik) [inline .gitmodules]: I believe travis allows running bash scripts. I will look into that next.

**Comment by dashersw** [comment]
We should be using environment variables.

**Comment by nilguncelik** [comment]
Ok, I changed .travis.yml accordingly and force pushed.

**Pull request merged**